### PR TITLE
feat(rook): multi package airgap bundles

### PR DESCRIPTION
**Pull request by emosbaugh**
For a better user experience when upgrading many versions of add-ons at once, enable multi-package bundles built on the fly.

Adds the following routes

```
/bundle/{installerID}/packages/{packages}
/bundle/version/{kurlVersion}/{installerID}/packages/{packages}
```

For example

```
curl -LO https://kurl.sh/bundle/879c9f9/packages/rookupgrade-10to14,rook-1.4.9,rook-1.5.12,rook-1.6.11,rook-1.7.11,rook-1.8.10,rook-1.9.12,rook-1.10.8.tar.gz
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -46,19 +46,6 @@ run-debug:
 archive-modules:
 	tar cfz node_modules.tar.gz node_modules/
 
-.PHONY: build-staging
```
Comment (emosbaugh) [inline Makefile:49]: this is unused

File: src/controllers/Bundles.ts [hunk 2]
```diff
@@ -42,60 +51,64 @@ export class Bundle {
...
       return notFoundResponse;
     }
 
     const externalVersions = getExternalAddonVersions();
-    installer = installer.resolve(installerVersions, externalVersions);
+
+    // resolve the correct version if specified in the spec
     kurlVersion = kurlVersionOrDefault(kvarg, installer);
     installerVersions = await getInstallerVersions(this.distURL, kurlVersion);
+    installer = installer.resolve(installerVersions, externalVersions);
```
Comment (laverya) [inline src/controllers/Bundles.ts:94]: I have a sneaking suspicion this fixes the versioned airgap bundle issue 😄

File: pkg/bundle/handler.go [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+
+	kurlVersion := vars["kurlVersion"]
+	installerID := vars["installerID"]
+	pkgs := strings.Split(strings.TrimSuffix(vars["packages"], ".tar.gz"), ",")
+
+	bundle, err := h.backend.GetBundlePackages(r, kurlVersion, installerID, pkgs)
+	if err != nil {
+		handleBundleError(err, w, r)
+		return
+	}
```
Comment (laverya) [inline pkg/bundle/handler.go:90]: I still feel like we should be able to get the list of packages without hitting typescript at all here

not worth arguing over though

Comment (laverya) [reply pkg/bundle/handler.go:90]: AFAICT the only scenario in which this adds information is when there's no kurlVersion specified in the URL

**Review by laverya — APPROVED** [review]

**Review by rrpolanco — APPROVED** [review]

**Pull request merged**